samcli/lib/init/template_modifiers/application_insights_template_modifier.py

```diff
@@ -0,0 +1,95 @@
+"""
+Class used to parse and update template when application-insights is enabled
+"""
+import logging
+from typing import Any
+from samcli.lib.init.template_modifiers.cli_template_modifier import TemplateModifier
+
+LOG = logging.getLogger(__name__)
+
+
+class ApplicationInsightsTemplateModifier(TemplateModifier):
+    import ruamel.yaml
+    from ruamel.yaml import YAML
+    from ruamel.yaml.comments import CommentedMap
+
+    TYPE_KEY = "Type"
+    RESOURCES_KEY = "Resources"
+    PROPERTIES_KEY = "Properties"
+    NAME_KEY = "Name"
+    RESOURCE_QUERY_KEY = "ResourceQuery"
+    RESOURCE_GROUP_NAME_KEY = "ResourceGroupName"
+    AUTO_CONFIG_ENABLED_KEY = "AutoConfigurationEnabled"
+    DEPENDS_ON_KEY = "DependsOn"
+    RESOURCE_GROUP_TYPE = "AWS::ResourceGroups::Group"
+    CFN_STACK_TYPE = "CLOUDFORMATION_STACK_1_0"
+    APPLICATION_INSIGHTS_TYPE = "AWS::ApplicationInsights::Application"
+    RESOURCE_GROUP_REF = "ApplicationResourceGroup"
+    APPLICATION_INSIGHTS_REF = "ApplicationInsightsMonitoring"
+    AUTO_CONFIG_VALUE = "true"
+    RESOURCE_GROUP_NAME = {"Fn::Join": ["", ["ApplicationInsights-SAM-", {"Ref": "AWS::StackName"}]]}
```

Comment:
Will this create any issues if resource group name has size limitations, if how can we eliminate it?

Reply:
A stack name cannot be longer than 128 chars. Size limit of ResourceGroup name is 300 chars. So, that shouldn't be a problem